src/gamestate/system_state.cpp

```diff
@@ -986,19 +986,39 @@ dcon::trigger_key state::commit_trigger_data(std::vector<uint16_t> data) {
 }
 
 dcon::effect_key state::commit_effect_data(std::vector<uint16_t> data) {
-	if(data.empty())
- 		return dcon::effect_key();
-	
-	if(effect_data.empty()) { // Create placeholder for invalid effects
-		effect_data.push_back(uint16_t(effect::no_payload));
+	if(data.empty()) {
+		if(effect_data_indices.empty())
+			effect_data_indices.push_back(int32_t(0));
+		return dcon::effect_key();
 	}
 
-	auto start = effect_data.size();
-	auto size = data.size();
+	if(effect_data_indices.empty()) { // Create placeholder for invalid effects
+		effect_data_indices.push_back(0);
+		effect_data.push_back(uint16_t(effect::no_payload));
+	}
 
-	effect_data.resize(start + size, uint16_t(0));
-	std::copy_n(data.data(), size, effect_data.data() + start);
-	return dcon::effect_key(uint16_t(start));
+	auto search_result = std::search(effect_data.data(), effect_data.data() + effect_data.size(),
```

Comment:
Any data on how often this actually finds something? While for triggers, there are lots of duplicate conditions (like all the triggers that check for a specific tag) I wasn't expecting there to be a lot of duplicate events, and looking through all the effect data for each effect can be slow

Reply:
for TGC it's around 32K, and effects are somewhere around a hundred or so

Reply:
You mean that there are a hundred duplicate effects? If so, that may not be worth the search time

Reply:
Pardon me, on vanilla there are 260 duplicate effects, on TGC there are around 600